### ISSUE TITLE
[kirby-3.9.0-rc] Rendering pages inside of slots messes up slot content

### ISSUE THREAD

**Issue by rasteiner** [k-next/kirby#4997]
## Description
Using `$page->render(contentType: 'json')` (or equivalent) inside of a slot discards all previous content passed to that slot and puts the content after the call after the snippet. 

**Expected behavior**  
It shouldn't matter what I pass into the slot. 

## To reproduce
```php
<?php // site/snippets/container.php ?>

<!-- snippet begins -->
<div>
    <!-- in snippet before slot mount -->
    <?= $slot ?>
    <!-- in snippet after slot mount -->
</div>
<!-- snippet ends -->
```

```php
<?php // site/templates/default.php ?>

<!-- before snippet call -->
<?php snippet('container', slots: true) ?>
   <!-- before render -->
   <?= $page->render(contentType: 'json') ?>
   <!-- after render -->
<?php endsnippet() ?>
<!-- after snippet call -->
```

```php
<?php // site/templates/default.json.php ?>

[1, 2, 3]
```

The currently rendered output is (formatted for clearness):
```html
<!-- before snippet call -->
<!-- snippet begins -->
<div>
    <!-- in snippet before slot mount -->
    [1, 2, 3]
    <!-- in snippet after slot mount -->
</div>
<!-- snippet ends -->
<!-- after render -->
<!-- after snippet call -->
```
Notice `<!-- before render -->` has disappeared, and `<!-- after render -->` is in the wrong place. 

## Your setup
**Kirby Version**  
3.9.0-rc.2

**Additional context**  
I guess it has something to do with output buffering? Strangely tough creating an example with `ob_start()` and `ob_get_clean()` does *not* show this behaviour. 

I'm probably slightly abusing `$page->render()` here (and in my specific case I can work around the problem by moving that call outside of the slot), the issue might however be a symptom of a deeper problem.

**Comment by lukasbestle** [comment]
@rasteiner Thanks for testing and your detailed report. I found the issue, it's fortunately not a deep problem with output buffering. I think we'll be able to fix the bug before the 3.9 release.

**Comment by bastianallgeier** [comment]
✅
